trin-core/src/portalnet/overlay_service.rs

```diff
@@ -68,6 +68,7 @@ use ssz::Encode;
 use ssz_types::{BitList, VariableList};
 use thiserror::Error;
 use tokio::sync::mpsc::{self, UnboundedReceiver, UnboundedSender};
+use tokio::sync::Mutex;
```

Comment:
I might be wrong here, but re: `"asynchronous mutex would only be necessary if we need to hold a lock across await points"` ... isn't that exactly what we're doing [here](https://github.com/ethereum/trin/pull/373/files#diff-354721dfd15f4b0d0f5df418d0ac4118592bf96bda720655548e380faf9f1b37R1368)?

`I'm not sure why that method takes &mut self in the first place` - agreed, but updating the self mut ref on validator and changing the `Mutex` to a `RwLock` doesn't change that since `validator.validate_content().await` is async, and a future created by an async block is not `Send`, which is a required trait by `tokio::spawn`. lmk if I'm missing something about this situation.

Reply:
Ah yes, so I was wrong above on that. But on the second point, if we change the validator to require `&self` instead of `&mut self`, then we can just wrap the validator in an `Arc` (without a lock) which is `Send`.

Looking at the [`ChainHistoryValidator`](https://github.com/ethereum/trin/blob/d6d8f722cb773ac4ca51874429072d49dc363ea4/trin-history/src/validation.rs#L23), I don't see why we need a `&mut self` receiver.